### PR TITLE
refactor: Replace machinery with tasqueue

### PR DESCRIPTION
**Pull request by kalbhor**
This MR aims to replace the underlying job library (machinery) with tasqueue. Apart from internal changes that do not affect this libraries' clients, the following changes/suggestions are made: 

- Manage the runtime cleanly via contexts. Currently I've passed a simple context to the `Start()` method. We can discuss how to handle the cancellation (this is omitted as of now).

- Change queue/concurrency configs. Tasqueue expects the concurrency parameter to be passed when registering the task (with some certain queue), unlike the existing way where concurrency is being passed when a group job is supposed to be queued. (We can discuss how to make this cleaner for clients) 

- Log management: Tasqueue uses `zerodha/logf` internally, maybe we can use this across the project 

- Ordering of pending jobs: The current testcases assume that the pending jobs returned will be in the same order as submission. This is implemented in a reverse manner in Tasqueue

- Change in config: Instead of using DSNs to pass configs, have added a declarative config for both the results and broker options for Tasqueue.

PS: Any readme related changes can be made once the MR is finalized

### CODE REVIEW

**Comment by knadh** [comment]
This is excellent. The Machinery build is 26MB. The Tasqueue build is 7.9MB :tada: 

Hasn't the new stdlib `slog` superseded `zerodha/logf`? @mr-karan?

**Comment by mr-karan** [comment]
> Hasn't the new stdlib slog superseded zerodha/logf?

Yes, we plan to support slog backend in `zerodha/logf` eventually but the API will be `slog` itself. Better to use that here as we can switch backends later but have a consistent logging API.

**Comment by knadh** [comment]
@kalbhor let's replace the `zerodha/logf`  with `slog` in Tasqueue?

**Comment by kalbhor** [comment]
Have replaced `zerodha/logf` with `slog` in Tasqueue. It will accept an appropriate `slog.Handler` (which can be swapped later if needed, like mr-karan has mentioned). Made these changes in DungBeetle as well, as a part of this PR

File: cmd/config.sample.toml [hunk 1]
```diff
@@ -1,18 +1,28 @@
-# Redis addresses are denoted as: redis://password@host:port/db_num
-
-[job_queue]
-broker_address = "redis://127.0.0.1:6379/1"
-
-# This is where Machinery stores it's job states.
-# In Machinery's terminology, this is its "result backend"
+[job_queue.broker]
+address = ["localhost:6379"]
```
Comment (knadh) [inline cmd/config.sample.toml:2]: `addrs` or `addresses` (plural) would be make is clearer.

File: cmd/config.sample.toml [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
-# where DungBeetle stores results of jobs, bypassing machinery.
-state_address = "redis://127.0.0.1:6379/1"
-
-# TTL in seconds after which a job entry and its meta disappears from the Machinery
-# store. This does not remove results from the results DB.
-state_ttl = "30m"
-
+# where DungBeetle stores results of jobs, bypassing tasqueue.
+[job_queue.results]
+address = ["localhost:6379"]
```
Comment (knadh) [inline cmd/config.sample.toml:16]: Same here.

File: cmd/config.sample.toml (hunk 1 above)
Comment (kalbhor) [reply cmd/config.sample.toml]: Yepp, using `addresses` since we're not using shorthands elsewhere (`password` / `pass`, etc)

**Pull request merged**